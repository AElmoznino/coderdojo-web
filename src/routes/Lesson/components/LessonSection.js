// @flow

import React from 'react'
import ReactMarkdown from 'react-markdown'
import H1 from 'components/Typography/H1'

type LessonSectionProps = {
  sectionBody: string,
  sectionTitle: string,
  sectionImgAlt?: string,
  sectionImgUrl?: string,
  sectionFinishingText?: string,
  sectionSandbox?: string,
}

export const LessonSection = ({
  sectionBody,
  sectionTitle,
  sectionImgAlt,
  sectionImgUrl,
  sectionFinishingText,
  sectionSandbox,
}: LessonSectionProps) => (
  <div>
    <H1>{sectionTitle}</H1>

    <ReactMarkdown>{sectionBody}</ReactMarkdown>

    {sectionImgUrl && <img alt={sectionImgAlt} src={sectionImgUrl} />}

    {sectionSandbox && (
      <iframe
        src={sectionSandbox}
        style={{
          width: '100%',
          height: '500px',
          overflow: 'hidden',
        }}
        sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
        title="unique"
      />
    )}

    {sectionFinishingText && <p>{sectionFinishingText}</p>}
  </div>
)

export default LessonSection
