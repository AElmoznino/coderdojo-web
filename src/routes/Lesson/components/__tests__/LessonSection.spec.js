import React from 'react'
import { shallow } from 'enzyme'
import LessonSection from '../LessonSection'

describe('components/LessonSection', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <LessonSection sectionBody="Test body" sectionTitle="Test Title" />,
    )
  })

  it('renders LessonSection', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders LessonSection with image', () => {
    component.setProps({
      sectionImgAlt: 'Test image alt',
      sectionImgUrl: 'Test image url',
    })

    expect(component).toMatchSnapshot()
  })

  it('renders LessonSection with sandbox and explaining text after it', () => {
    component.setProps({
      sectionFinishingText: 'Test text explaining the sandbox above it',
      sectionSandbox: 'https://codesandbox.io/embed/some-test-sandbox',
    })

    expect(component).toMatchSnapshot()
  })
})
