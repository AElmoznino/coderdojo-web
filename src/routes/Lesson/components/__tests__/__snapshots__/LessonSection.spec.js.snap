// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LessonSection renders LessonSection 1`] = `
<div>
  <styled.h1>
    Test Title
  </styled.h1>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={true}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    sourcePos={false}
    transformLinkUri={[Function]}
  >
    Test body
  </ReactMarkdown>
</div>
`;

exports[`components/LessonSection renders LessonSection with image 1`] = `
<div>
  <styled.h1>
    Test Title
  </styled.h1>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={true}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    sourcePos={false}
    transformLinkUri={[Function]}
  >
    Test body
  </ReactMarkdown>
  <img
    alt="Test image alt"
    src="Test image url"
  />
</div>
`;

exports[`components/LessonSection renders LessonSection with sandbox and explaining text after it 1`] = `
<div>
  <styled.h1>
    Test Title
  </styled.h1>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={true}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    sourcePos={false}
    transformLinkUri={[Function]}
  >
    Test body
  </ReactMarkdown>
  <iframe
    sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
    src="https://codesandbox.io/embed/some-test-sandbox"
    style={
      Object {
        "height": "500px",
        "overflow": "hidden",
        "width": "100%",
      }
    }
    title="unique"
  />
  <p>
    Test text explaining the sandbox above it
  </p>
</div>
`;
